src/main/java/net/bhardy/bizzo/billing/impl/ActionChoiceBuilder.java

```diff
@@ -44,21 +45,18 @@
     @Override
     public FilterOption action(Kind actionKind) {
         BillingPolicy layered = new BillingPolicy() {
+            private static final int RANGE = 12;
+
             @Override
             public boolean isDueOn(final LocalDate today) {
-                if (!filter.applies(today)) {
-                    return false;
-                }
-                if (underlyingPolicy.isDueOn(today)) {
-                    return true;
-                }
-                LocalDate original = findOriginalDate(today, actionKind);
-                return underlyingPolicy.isDueOn(original);
+                final Period howFarBack = underlyingPolicy.getCycleType().getPeriod().multipliedBy(RANGE);
+                final LocalDate jumpBack = today.minus(howFarBack);
+                return upcomingDueDates(jumpBack).limit(RANGE * 2).anyMatch(today::equals);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Cast one of the operands of this multiplication operation to a "long". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2184)